### PR TITLE
Manual Jasper 4 migration

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Is there a reason why jasper cannot be unpinned. i think there will be conflicts in the ecosystem without this.

**Comment by hmaarrfk** [comment]
for more context, this is currently a blocker for part of the jasper4 migration.

**Comment by dopplershift** [comment]
@zklaus You pinned jasper in #152, any thoughts?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,8 +22,8 @@ requirements:
     - cmake
     - make
   host:
-    - jasper >=2.0.14,<3  # [not ppc64le]
-    - openjpeg  # [ppc64le]
```
Comment (dopplershift) [inline recipe/meta.yaml:27]: This previously only used `openjpeg` on ppc64le systems, and did not use jasper in that case. Now this unconditionally depends on both. That seems wrong?

Comment (hmaarrfk) [reply recipe/meta.yaml:27]: i can revert. but that seems wrong too right?

Comment (dopplershift) [reply recipe/meta.yaml:27]: Well, I'd say looking at the build logs to understand what it's now doing on both platforms and seeing if it's correct or if one of them is unnecessary would be the right thing to do.

Comment (hmaarrfk) [reply recipe/meta.yaml:27]: ok, it seems that some previous tests were failing and it was explicitely disabled in the build. sorry for not tkaing a closer look.

Comment (iainrussell) [reply recipe/meta.yaml:27]: I think the next release of ecCodes will fix that failing test, or at least not run it with that specific combination of jasper+threads (am I right, @shahramn?)

Comment (hmaarrfk) [reply recipe/meta.yaml]: I've reverted the changes to openjpeg.

**Comment by hmaarrfk** [comment]
I mostly just wanted to bring to the attention that the bot thinks that jasper4 is migrated on this feedstock, but in reality this feedstock is still built with jasper 2.

This is causing some downstream packages to not get migrated with jasper correctly.

image


Sorry for trying to bundle this with some openjpeg "improvements". That was an over reach.

From reading #153 it seems that mostly it was a faulty pinning that cause the incompatibility, but I may be wrong.

**Comment by zklaus** [comment]
> @zklaus You pinned jasper in #152, any thoughts?

I pinned Jasper because in Jasper 3 the approach to multi-threading was changed.  This was adapted in eccodes, but unfortunately faultily. Thus, eccodes with multi-threading was incompatible with Jasper >=3. I don't think that has changed.

**Comment by hmaarrfk** [comment]
hmm. ok, i can't find some open source forum for bug reports.

I did find a note about compilation fixes with jasper 4, but nothing explicitly mentions  multi threading in the 2 changelogs i  searched for
https://confluence.ecmwf.int/display/ECC/ecCodes+version+2.28.0+released

**Comment by ocefpaf** [comment]
Should we pull the packages here?

**Pull request merged**